micrometer-core/src/test/java/io/micrometer/core/instrument/simple/SimpleMeterRegistryTest.java

```diff
@@ -101,6 +109,133 @@ public void newFunctionCounterWhenCountingModeIsStepShouldReturnStepFunctionCoun
+                .sla(Duration.ofMillis(20), Duration.ofMillis(50), Duration.ofMillis(100))
+                .publishPercentileHistogram()
+                .register(registry);
+
+        final AtomicLong testRunTime = new AtomicLong();
+        TimeGauge
+                .builder("test.run.time", testRunTime, TimeUnit.NANOSECONDS, AtomicLong::doubleValue)
+                .register(registry);
+
+        FunctionTimer
+                .builder("cache.latency", new Object(), o -> 10, o -> 300, TimeUnit.MILLISECONDS)
+                .register(registry);
+
+        FunctionCounter
+                .builder("cache.evictions", new Object(), o -> 10)
+                .baseUnit("beans")
+                .register(registry);
+
+        final LongTaskTimer taskTimer = LongTaskTimer
+                .builder("test.loop.timer")
+                .register(registry);
+
+        Meter.builder("default.meter", Type.GAUGE,
+                Arrays.asList(
+                        new Measurement(() -> 7.7, Statistic.VALUE),
+                        new Measurement(() -> 8.8, Statistic.VALUE)
+                ))
+                .baseUnit("beans")
+                .register(registry);
+
+        final LongTaskTimer.Sample loopSample = taskTimer.start();
+
+        for (int i = 0; i < 300; i++) {
+            responseTimer.record(Duration.ofMillis(i));
+            requestSizeCounter.increment(i);
+            responseSizeSummary.record(i);
+        }
+
+        loopSample.stop();
+
+        testRunTime.set(200);
+        assertThat(registry.toString()).isEqualTo("\n"
+                + "Counter cache.evictions count=10.000000\n"
+                + "Timer cache.latency\n"
+                + "        count = 10.000000\n"
+                + "    totalTime = 0.300000\n"
+                + "         mean = 0.030000\n"
+                + "Gauge default.meter\n"
+                + "        VALUE = 7.700000\n"
+                + "        VALUE = 8.800000\n"
+                + "Counter request.size count=44850.000000\n"
+                + "DistributionSummary response.size\n"
+                + "        count = 300\n"
+                + "        total = 44850.000000\n"
+                + "         mean = 149.500000\n"
+                + "          max = 299.000000\n"
+                + "    percentileValues:\n"
+                + "        75.0 <= 231.9375\n"
+                + "        90.0 <= 271.9375\n"
+                + "        95.0 <= 287.9375\n"
+                + "        99.0 <= 303.9375\n"
+                + "       99.99 <= 303.9375\n"
+                + "    histogramCounts:\n"
+                + "       250.0 <= 256.0\n"
+                + "       500.0 <= 300.0\n"
+                + "      1000.0 <= 300.0\n"
+                + "      2000.0 <= 300.0\n"
+                + "      4000.0 <= 300.0\n"
+                + "      6000.0 <= 300.0\n"
+                + "Gauge response.size.histogram {le=250} value=256.000000\n"
```

Comment:
It seems like it'd be nice if we could deduplicate these gauges since this info is already present in the above histogramCounts.